Improve benchmarks

squiddy opened the pull request:
```
Each generator was passed a empty chunk for benchmarking. This didn't
reflect the whole complexity of some generators. They stopped early
because conditions where not met.
Now a chunk passed to a generator is prepopulated by the plugins that
are specified in the before attribute. Both complex and boring
generators are exempt, the former takes too long to run and the latter
is too simple compared to simplex.
```

So for example, the grass generator gets a chunk that was prepopulated by simplex and erosion generator.

I don't really like the changes to _sequential_seeded()_ and _repeated_seeds()_ (probably can be done nicer), but somehow the prepopulation has to be exempt from the benchmarking. Let's see this as a base for discussion. I'm not that good at names, so they should be changed too I think.

Besides, I added a simple benchmark for the season plugins to see the affect of a previous patch and the use of _set_block()_.

squiddy committed:
benchmarks: make them more practical with prepopulated chunks

Each generator was passed a empty chunk for benchmarking. This didn't
reflect the whole complexity of some generators. They stopped early
because conditions where not met.
Now a chunk passed to a generator is prepopulated by the plugins that
are specified in the before attribute. Both complex and boring
generators are exempt, the former takes too long to run and the latter
is too simple compared to simplex.

squiddy committed:
benchmarks: added benchmarks for season plugins

MostAwesomeDude commented:
The benchmarks never really got wired up to a codespeed or anything like that. If you wanna put forward patches for them, go for it, but keep in mind that their accuracy and whatnot is not really testable. IOW, check your code carefully.

squiddy committed:
benchmarks: remove seasons benchmark

The change to set_block (exiting early when chunk is not populated) made
it look like the performance of seasons improved, when in fact it will
perform just as good/bad as before. The benchmark passed a unpopulated chunk
to the transform function, whereby a running server will already have
the populated flag set.

squiddy commented:
Sorry, I'm not sure I understand you correctly. What do you mean by 'never got wired up to a codespeed'?

Now that I've seen the _chunkbench.py_ and _simplexbench.py_ in _tools/_, where at least chunkbench seems to mirror the functionality of benchmarks/chunks.py, I'm wondering what's the plan for these? The scripts under tools/ are from december last year, benchmarks/ from january this year.

MostAwesomeDude commented:
https://github.com/tobami/codespeed/ is codespeed, and http://speed.pypy.org/ is what it looks like when properly set up.

Yeah, the tools were copied to benchmarks in preparation for some kind of unified benchmark, but never finished. You may have also noticed run_benchmarks. I'm really not sure what the best course of action is.
